### PR TITLE
Dedupe results, order by date posted, and remove sponsored results

### PR DESCRIPTION
**Pull request by mdentremont**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -10,14 +12,15 @@ var config = require('./config');
...
-    constructor(url, image, title, description, location, price) {
+    constructor(url, image, title, description, location, price, datePostedText) {
         this.url = url;
         this.image = image;
         this.title = title;
         this.description = description;
         this.location = location;
         this.price = price;
         this.isBusiness = null;
+        this.datePosted = Ad.determineDatePosted(datePostedText);
```
Comment (DanLatimer) [inline server.js]: let's make this just this.datePosted = datePosted and do the determineDatePosted only in one spot

File: server.js [hunk 2]
```diff
@@ -85,16 +113,20 @@ function createAdFetchPromise(url) {
     return new RSVP.Promise((resolve, reject) => {
         request(url, (error, response, html) => {
             const $ = cheerio.load(html);
+
             const parsedAds = $('div.search-item').get()
-                .map(item => Ad.buildAd($(item)));
+                .map(item => Ad.buildAd($(item)))
+                .filter(ad => ad.datePosted != null); // removes sponsored ads
```
Comment (DanLatimer) [inline server.js]: this will also remove all ads if they ever change their dom....

File: server.js [hunk 3]
```diff
@@ -29,10 +32,35 @@ class Ad {
...
+        if (datePostedText === 'Yesterday') {
+            return moment().subtract(1, 'day');
+        }
+
+        var parsedDateString = moment(datePostedText, 'DD-MM-YYYY');
+        if (parsedDateString.isValid()) {
+            return parsedDateString;
+        }
+
+        console.log("Unexpected date posted text:" + datePostedText);
```
Comment (DanLatimer) [inline server.js]: maybe elevate this to error

File: server.js (hunk 1 above)
Comment (mdentremont) [inline server.js]: Done

File: server.js (hunk 3 above)
Comment (mdentremont) [inline server.js]: Done

File: server.js (hunk 2 above)
Comment (mdentremont) [inline server.js]: I suppose we can live with sponsored ads for meow

**Pull request merged**